docs_src/design/adr/system-management/0006-Metrics-ControlPlaneEvents.md

```diff
@@ -0,0 +1,239 @@
+- Count of service start and restart attempts
+- **CPE** service stop or restart
+- **CPE** registration of new devices and services (allowing for future quarantine mechanism)
+- **CPE** indication of any new uploading of new CA/import to vault
+
+### Design Proposal
+
+- All services will use/integrate go-mod-messaging (if not already integrated).
+- A new metric/CPE model/DTO (Telemetry/Bucket) is required for services to structure the metric or CPE data.  The model/DTO will have similarities to the core event/reading model/DTO.
+    - Even though metric and CPE manifest from different circumstances (a scheduled collection versus monitored event in the service), the same model/DTO should be used to represent either.  This simplifies both the publish and subscribe to metrics/CPE data.
+    - *Should we use the same Event/Reading structure for metric and CPE messages?  The event/reading structure has some elements (like device and binary value) that would not make sense in a metric/CPE message.  Additionally, the metric or CPE would want to be attributed to the service that created it.*
+    - The proposed message structure (DTO) for metrics and CPE (for inclusion in go-mod-core-contracts) is:
+  
+  ``` go
+    type Telemtry struct {
+        Id          string            `json:"id,omitempty" codec:"id,omitempty"`
+        Service     string            `json:"device,omitempty" codec:"device,omitempty"`  // originating service
+        Created     int64             `json:"created,omitempty" codec:"created,omitempty"`
+        Modified    int64             `json:"modified,omitempty" codec:"modified,omitempty"`
+        Buckets    []Bucket         `json:"readings,omitempty" codec:"buckets,omitempty"`
+        Tags        map[string]string `json:"tags,omitempty" codec:"tags,omitempty" xml:"-"`
+    }
+
+    type Bucket struct {
+        Id            string `json:"id,omitempty" codec:"id,omitempty"`
+        Created       int64  `json:"created,omitempty" codec:"created,omitempty"` 
+        Modified      int64  `json:"modified,omitempty" codec:"modified,omitempty"`
+        Service       string `json:"device,omitempty" codec:"device,omitempty"`
+        Name          string `json:"name,omitempty" codec:"name,omitempty"`  // metric or CPE name key
+        Value         string `json:"value,omitempty" codec:"value,omitempty"` // metric or CPE value
+        ValueType     string `json:"valueType,omitempty" codec:"valueType,omitempty"`
+        Description   string `json:"name,omitempty" codec:"description,omitempty"`  // human readable details
+    }
+  ```
+
+- Proposed endpoint for the REST endpoints that respond with what metrics and CPE the service offers are:  /api/v2/metrics and /api/v2/cpe
+- Configuration, not unlike that provided in core data, will specify what message endpoints the metrics and CPE messages should be sent.
+    - Proposed configuration for each service for metric and CPE endpoints:
+
+  ``` yaml

```

Comment:
Is that just an implementation detail? Why wouldn't you use JSON especially when we're talking about REST endpoints?

Reply:
proposed just to give example for clarification.